bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDCyclicTransposeFlowGraphFinder.java

```diff
@@ -0,0 +1,184 @@
+     * @param sink Single sink vertex without previous vertices calculated
+     * @param inputPins Relevant input pins on the given vertex
+     * @return List of sinks created from the initial sink with previous vertices calculated
+     */
+    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> inputPins, List<Node> sourceNodes) {
+        List<DFDVertex> vertices = new ArrayList<>();
+        vertices.add(sink);
+
+        if (sourceNodes.contains(sink.getReferencedElement())) {
+            return vertices;
+        }
+
+        for (Pin inputPin : inputPins) {
+
+            List<Flow> incomingFlowsToPin = dataFlowDiagram.getFlows()
+                    .stream()
+                    .filter(flow -> flow.getDestinationPin()
+                            .equals(inputPin))
+                    .toList();
+
+            List<String> previousNodesInTransposeFlow = new ArrayList<>();
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getDestinationNode()
+                    .getEntityName());
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName());
+
+            List<DFDVertex> finalVertices = vertices;
+            vertices = (incomingFlowsToPin.stream()
+                    .flatMap(flow -> handleIncomingFlow(flow, inputPin, finalVertices, sourceNodes, previousNodesInTransposeFlow).stream())
+                    .toList());
+        }
+        return vertices;
+    }
+
+    private List<DFDVertex> determineSinks(DFDVertex sink, List<Pin> inputPins, List<Node> sourceNodes, List<String> previousNodesInTransposeFlow) {
+        List<DFDVertex> vertices = new ArrayList<>();
+        vertices.add(sink);
+
+        if (sourceNodes.contains(sink.getReferencedElement())) {
+            return vertices;
+        }
+
+        for (Pin inputPin : inputPins) {
+            List<Flow> incomingFlowsToPin = dataFlowDiagram.getFlows()
+                    .stream()
+                    .filter(flow -> flow.getDestinationPin()
+                            .equals(inputPin))
+                    .toList();
+
+            if (!LoopCheck(previousNodesInTransposeFlow, incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName()))
+                continue;
+
+            previousNodesInTransposeFlow.add(incomingFlowsToPin.get(0)
+                    .getSourceNode()
+                    .getEntityName());
+
+            List<DFDVertex> finalVertices = vertices;
+            vertices = (incomingFlowsToPin.stream()
+                    .flatMap(flow -> handleIncomingFlow(flow, inputPin, finalVertices, sourceNodes, previousNodesInTransposeFlow).stream())
+                    .toList());
+        }
+        return vertices;
+    }
+
+    public List<DFDVertex> handleIncomingFlow(Flow incomingFlow, Pin inputPin, List<DFDVertex> vertices, List<Node> sourceNodes,
```

Comment:
Should this method be public? 
Documentation is missing

Reply:
added documentation